bundles/org.openhab.persistence.rrd4j/src/main/java/org/openhab/persistence/rrd4j/internal/RRD4jPersistenceService.java

```diff
@@ -320,8 +317,8 @@ public Iterable<HistoricItem> query(FilterCriteria filter) {
 
             // do not call method {@link RrdDb#createFetchRequest(ConsolFun, long, long, long)} if start > end to avoid
             // an IAE to be thrown
-            if (start > end) {
-                logger.warn("Could not query rrd4j database for item '{}': start ({}) > end ({})", itemName, start,
+            if (start > end && filter.getEndDate() == null) {
```

Comment:
Logic is now wrong: If `start > end && filter.getEndDate() != null`, we will now call the method and fully reintroduce the bug fixed by #14238. You probably meant something like this:
```java
if (start > end) {
    if (filter.getEndDate() != null)
        logger.debug("Could not query rrd4j database for item '{}': start ({}) > end ({})", itemName, start, end);
    }
    return List.of();
}
```

Reply:
hm, are you sure? - if end date is not set (which is the case when invoking API with `boundary=true` and we "calculate" end timestamp - only in this case we simply return empty response and don't trigger an exception. Tested locally and it works. If user would invoke a query with explicitly setting end time and `end > start`, "old" logic would kick in. To be honest I'm not sure if we should add (at all) `&& filter.getEndDate() == null` part,  ... and maybe only check `start > end` ... but then explicit queries would also only log `debug` and return empty response. Open for suggestions ...

Also checked a bit the code and it seems datapoints are stored using `long now = System.currentTimeMillis() / 1000;` - https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.persistence.rrd4j/src/main/java/org/openhab/persistence/rrd4j/internal/RRD4jPersistenceService.java#L189 so there should be no datapoint "in the future" so having 
```
long end = filter.getEndDate() == null ? System.currentTimeMillis() / 1000
```
should probably be good enough. We might also use `getLastUpdateTime()` but not sure if that would bring much benefit ...

Reply:
Please remember - when rendering chart with i.e.
```
http://<ip>/rest/persistence/items/<item>?starttime=2022-12-31T23:00:00.000Z&endtime=2023-01-31T22:59:59.999Z&boundary=true
```
will query database three times (due `boundary=true`):
1 . for a point before `starttime`, 
2. with explicit start/end- `starttime=2022-12-31T23:00:00.000Z` and `endtime=2023-01-31T22:59:59.999Z`,
3. start time of query = end time (2023-01-31T22:59:59.999Z in above example) and no end time set - which is the offending query before the fix and raised an exception. Therefore `filter.getEndDate() == null` since no end is set in this case

Reply:
If `filter.getEndDate() == null`, `end` is set to `System.currentTimeMillis()` in line 283. This may be before `start`. This scenario is already covered by your code.

If `filter.getEndDate() != null` and `end` is before `start`, this early exit will not happen. Instead `createFetchRequest` will be called below and the exception is thrown.

Reply:
IMHO it should not be triggered (start > end) from UI unless there is a bug. The reason for adding this additional check was (a bit) based on your comment https://github.com/openhab/openhab-addons/pull/14238#discussion_r1082220561 so if user would invoke API and explicitly set `start > end` ... but I guess the exception that would be raised (and I guess it was in the past) is not "the best" ... maybe we should really just get rid of the `&& filter.getEndDate() == null` part?

Reply:
my rationale was - if user (or UI) did not specify end date via REST API call and internal logic results in `start > end` -> return empty list; if user did explicitly set `start > end` -> use existing logic (which triggers exception I guess, but did not try it out yet, will do so later today ...).

Reply:
Thanks for linking that comment of mine. I actually still agree with that (myself), I guess I'm just sleepy today. 🙂 In that case I would take the control then and explicitly throw that exception by ourselves instead of relying on external error handling. So we could validate parameters in the beginning of the method (pseudo code):
```java
ZonedDateTime filterBeginDate = filter.getBeginDate();
ZonedDateTime filterEndDate = filter.getEndDate()
if (filterBeginDate != null && filterEndDate != null && filterBeginDate.isAfter(filterEndDate)) {
    throw new IllegalArgumentException("begin (" + filterBeginDate + ") before filterEndDate (" + filterEndDate + ")");
}
```

Then we wouldn't have to deal with that later in the method.

Reply:
Great suggestion, updated accordingly, seems to work. Thx for your feedback, really appreciated.